lib/index.js

```diff
@@ -1 +1,86 @@
+var canvas = document.getElementById('game');
+var context = canvas.getContext('2d');
 
+function Fish(options) {
+  options = options || {};
+  this.x = options.x;
+  this.y = options.y;
+  this.width = options.width || 10;
+  this.height = options.height || 10;
+  this.velocity = options.velocity || 1;
+ }
+
+Fish.prototype.draw = function() {
+  context.fillRect(this.x, this.y, this.width, this.height);
+  return this;
+};
+
+Fish.prototype.move = function() {
+  this.x += this.velocity;
+  return this;
+};
+
+Fish.prototype.reverseDirection = function() {
+  if (this.x > canvas.width || this.x <= 0) {
+    this.velocity = -1 * this.velocity;
+    return this;
+  }
+};
+
+var fish1 = new Fish({x: 0, y: 200});
+var fish2 = new Fish({x: 100, y: 400});
+var fish3 = new Fish({x: 350, y: 500});
+var fish4 = new Fish({x: 550, y: 550, width: 50, height: 20, velocity: 0.5});
+
+var fishies = [fish1, fish2, fish3, fish4];
+
+function Boat(options) {
+  options = options || {};
+  this.x = options.x;
+  this.y = options.y;
+  this.width = options.width || 10;
+  this.height = options.height || 10;
+  this.velocity = options.velocity || 5;
+}
+
+Boat.prototype.draw = function() {
+  context.fillRect(this.x, this.y, this.width, this.height);
+  return this;
+};
+
+Boat.prototype.moveBoat = () => {
+  canvas.addEventListener('keydown', function(event) {
+    if (event.which === 37) {
+      console.log('left key stroke');
+      Boat.prototype.moveBoatLeft();
+    } else if (event.which === 39) {
+      Boat.prototype.moveBoatRight;
+    }
+  });
+};
+
+Boat.prototype.moveBoatLeft = function() {
+  console.log(this.x)
```

Comment:
@rrgayhart @stevekinney
- Currently returning canvas instead of Boat object as this. 
- Probably happens after addEventListener on line 52
- Worked on this for a couple hours, how do I get around it? Ideally, boat should move left when left key is pressed.